src/streaming/StreamProcessor.js

```diff
@@ -1002,7 +1019,53 @@ function StreamProcessor(config) {
         }
     }
 
-    function _handleInbandEvents(data, request, mediaInbandEvents, trackInbandEvents) {
+    function _handleInbandPrfts(isoFile, timescale) {
+        const prftBoxes = isoFile.getBoxes('prft');
+
+        const prfts = [];
+        prftBoxes.forEach(prft => {
+            prfts.push(_parsePrftBox(prft, timescale));
+        });
+
+        return prfts;
+    }
+
+    function _parsePrftBox(prft, timescale) {
+        // Get prft type according to box flags
+        let type = 'unknown';
+        switch (prft.flags) {
+            case 0:
+                type = DashConstants.PRODUCER_REFERENCE_TIME_TYPE.ENCODER;
+                break;
+            case 16: 
+                type = DashConstants.PRODUCER_REFERENCE_TIME_TYPE.APPLICATION;
+                break;
+            case 24: 
+                type = DashConstants.PRODUCER_REFERENCE_TIME_TYPE.CAPTURED;
+                break;
+            default:
+                break;
+        }
+
+        // Get NPT timestamp according to IETF RFC 5905, relative to 1/1/1900
+        let ntpTimestamp = (prft.ntp_timestamp_sec * 1000) + (prft.ntp_timestamp_frac / 2**32 * 1000);
+        ntpTimestamp = ntpToUTC(ntpTimestamp);
+
+        const mediaTime = (prft.media_time / timescale);
+
+        return {
+            type,
+            ntpTimestamp,
+            mediaTime
+        }
+    }
+
+    function ntpToUTC(ntpTimeStamp) {
```

Comment:
Suggest to move this to `TimelineConverter` or `Utils` in case it could be useful for other classes

Reply:
moved into src/streaming/utils/TimeUtils.js